pandas/core/indexes/multi.py

```diff
@@ -3095,6 +3103,31 @@ def _update_indexer(idxr, indexer=indexer):
         # empty indexer
         if indexer is None:
             return Int64Index([])._ndarray_values
+
+        # Generate tuples of keys by which to order the results
+        if need_sort:
```

Comment:
can you just check is_lexsorted?

Reply:
This is not the same thing. The index may or may not be lexsorted, but what I want to know here is if the keys given to .loc are in the same order has the index (see line 3058), and if not, I reorder the result in indexer to have them in a order reflecting the given keys order.